modules/indexer/code/git.go

```diff
@@ -161,15 +162,23 @@ func nonGenesisChanges(ctx context.Context, repo *repo_model.Repository, revisio
 		default:
 			log.Warn("Unrecognized status: %c (line=%s)", status, line)
 		}
-	}
 
-	cmd := git.NewCommand(ctx, "ls-tree", "--full-tree", "-l").AddDynamicArguments(revision).
-		AddDashesAndList(updatedFilenames...)
-	lsTreeStdout, _, err := cmd.RunStdBytes(&git.RunOpts{Dir: repo.RepoPath()})
-	if err != nil {
-		return nil, err
+		if (i%50 == 0 || i == len(lines)-1) && len(updatedFilenames) > 0 {
```

Comment:
Any reason to use 50 here? How to ensure it will not over the limit as we don't check the length of updatedFilenames?

Reply:
Hmm, I think it should work.

1. The limit of arguments length to execute subprocess is 32,767. See https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessa#parameters
2. The limit of file full name length is 260. See https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file#maximum-path-length-limitation
3. So, `50*260 = 13000 < 32767`, bingo!

However, there's some way to [remove the limit](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#registry-setting-to-enable-long-paths) of file full name length. No idea if it's necessary to handle it.

Whatever, this PR can make things better.

Reply:
I will change it to 30 since https://learn.microsoft.com/en-us/troubleshoot/windows-client/shell-experience/command-line-string-limitation#more-information said it's 8191. 8191 / 256 = 31.9 . So maybe 30 is a good digital.